runtime/autoload/provider/node.vim

```diff
@@ -5,35 +5,53 @@ let g:loaded_node_provider = 1
 
 let s:job_opts = {'rpc': v:true, 'on_stderr': function('provider#stderr_collector')}
 
+function! s:is_minimum_version(version, min_major, min_minor) abort
+  let nodejs_version = a:version
+  if !a:version
+    let nodejs_version = get(split(system(['node', '-v']), "\n"), 0, '')
+    if v:shell_error || nodejs_version[0] !=# 'v'
+      return 0
+    endif
+  endif
+  " [major, minor, patch]
+  let v_list = !!a:version ? a:version : split(nodejs_version[1:], '\.')
+  return len(v_list) == 3
+    \ && ((str2nr(v_list[0]) > str2nr(a:min_major))
+    \     || (str2nr(v_list[0]) == str2nr(a:min_major)
+    \         && str2nr(v_list[1]) >= str2nr(a:min_minor)))
+endfunction
+
 " Support for --inspect-brk requires node 6.12+ or 7.6+ or 8+
 " Return 1 if it is supported
 " Return 0 otherwise
-function! provider#node#can_inspect()
+function! provider#node#can_inspect() abort
   if !executable('node')
     return 0
   endif
-  let node_v = get(split(system(['node', '-v']), "\n"), 0, '')
-  if v:shell_error || node_v[0] !=# 'v'
+  let ver = get(split(system(['node', '-v']), "\n"), 0, '')
+  if v:shell_error || ver[0] !=# 'v'
     return 0
   endif
-  " [major, minor, patch]
-  let node_v = split(node_v[1:], '\.')
-  return len(node_v) == 3 && (
-  \ (node_v[0] > 7) ||
-  \ (node_v[0] == 7 && node_v[1] >= 6) ||
-  \ (node_v[0] == 6 && node_v[1] >= 12)
-  \ )
+  return (ver[1] ==# '6' && s:is_minimum_version(ver, 6, 12))
```

Comment:
Why are you not passing `split(ver[1:], '\.')`?
`s:is_minimum_version('v4.12.0', 6, 12)` returns 1

Reply:
Perhaps it's simpler to pass the raw version string and unconditionally split it to get the [major, minor,patch] version list.

Reply:
> Why are you not passing split(ver[1:], '\.')?

We already have too much similar code duplicated in various places (e.g. health/provider.vim). Was trying to avoid even more duplication.

Ultimately I would like to add a built-in function that offers this functionality.

Reply:
I prefer to cache the result of `node -v` instead. `:checkhealth` is slower now because `!my_string_variable` evaluates to true.

Reply:
Oops I just posted https://github.com/neovim/neovim/pull/7738